test/e2e/activator_test.go

```diff
@@ -0,0 +1,193 @@
+		"DeploymentScaledToZero",
+		test.ServingNamespace,
+		2*time.Minute); err != nil {
+		t.Fatalf("Failed waiting for deployment to scale to zero: %v", err)
+	}
+
+	endpoint, err := spoof.GetServiceEndpoint(clients.KubeClient.Kube)
+	if err != nil {
+		t.Fatalf("Could not get service endpoint for spoofing client")
+	}
+
+	url := fmt.Sprintf("http://%s/?timeout=%d", *endpoint, serviceSleep)
+
+	client, err := pkgTest.NewSpoofingClient(clients.KubeClient, logger, domain, test.ServingFlags.ResolvableDomain)
+
+	responseChannel := make(chan *spoof.Response, concurrency)
+	roundTrip := roundTrip(client, url)
+
+	sendRequests(roundTrip, concurrency, responseChannel, timeout, logger, t)
+
+	analyseResponses(responseChannel, concurrency, timeout, t)
+}
+
+func sendRequests(roundTrip func() (*spoof.Response, error), concurrency int, resChannel chan *spoof.Response, timeout time.Duration, logger *logging.BaseLogger, t *testing.T) {
+	var (
+		group     errgroup.Group
+		responses int32
+	)
+	doneChan := make(chan struct{})
+	timeoutChan := time.After(timeout)
+	errChan := make(chan error)
+
+	// Send out the requests asynchronously and wait for them to finish.
+	go func() {
+		logger.Info("Starting to send out the requests")
+		//	Send requests async and wait for the responses.
+		for i := 0; i < concurrency; i++ {
+			group.Go(func() error {
+				res, err := roundTrip()
+				if err != nil {
+					return fmt.Errorf("unexpected error sending a request, %v\n", err)
+				}
+				atomic.AddInt32(&responses, 1)
+				resChannel <- res
+				return nil
+			})
+		}
+		err := group.Wait()
+		if err != nil {
+			errChan <- fmt.Errorf("unexpected error making requests against activator: %v", err)
+		}
+		logger.Info("Done sending out requests")
+		doneChan <- struct{}{}
+	}()
+
+	logger.Info("Waiting for all requests to finish")
+
+done:
+	for {
+		select {
+		case <-doneChan:
+			break done
+		case <-time.Tick(5 * time.Second):
+			logger.Infof("Received responses: %d", atomic.LoadInt32(&responses))
+		case err := <-errChan:
+			t.Fatalf("Error happened while waiting for the responses: %v", err)
+		case <-timeoutChan:
+			t.Fatalf("Timed out after %s while collecting responses, collected responses %d, out of %d", timeout, atomic.LoadInt32(&responses), concurrency)
+		}
+	}
+	logger.Info("Finished waiting for the responses")
+
+	if atomic.LoadInt32(&responses) != int32(concurrency) {
+		t.Fatalf("Responses from the activator, wanted %d, got %d", responses, concurrency)
+	}
+}
+
+func analyseResponses(respChan chan *spoof.Response, total int, timeout time.Duration, t *testing.T) {
+	t.Helper()
+	timeoutChan := time.After(timeout)
+	wantResponse := http.StatusOK
+	for i := 0; i < total; i++ {
+		select {
+		case resp := <-respChan:
+			if resp != nil {
+				if resp.StatusCode != wantResponse {
+					t.Errorf("response code expected: %d, got %d", wantResponse, resp.StatusCode)
+				}
+			} else {
+				t.Errorf("no response code received for the request")
```

Comment:
```suggestion
				t.Error("No response code received for the request")
```
Not an `error`, so normal sentence.